src/Microsoft.Diagnostics.Monitoring.EventPipe/Counters/CounterPayload.cs

```diff
@@ -119,35 +144,44 @@ internal class RatePayload : CounterPayload
         }
     }
 
-    internal class PercentilePayload : CounterPayload
+    internal record struct Quantile(double Percentage, double Value);
+
+    internal sealed class PercentilePayload : MeterPayload
     {
-        public PercentilePayload(string providerName, string name, string displayName, string displayUnits, string metadata, IEnumerable<Quantile> quantiles, DateTime timestamp) :
-            base(providerName, name, metadata, 0.0, timestamp, "Metric", EventType.Histogram)
+        public PercentilePayload(string providerName, string name, string displayName, string displayUnits, string metadata, double value, DateTime timestamp) :
+            base(timestamp, providerName, name, displayName, displayUnits, value, CounterType.Metric, metadata, EventType.Histogram)
         {
             // In case these properties are not provided, set them to appropriate values.
             string counterName = string.IsNullOrEmpty(displayName) ? name : displayName;
             DisplayName = !string.IsNullOrEmpty(displayUnits) ? $"{counterName} ({displayUnits})" : counterName;
-            Quantiles = quantiles.ToArray();
         }
-
-        public Quantile[] Quantiles { get; }
     }
 
-    internal record struct Quantile(double Percentage, double Value);
-
-    internal class ErrorPayload : CounterPayload
+    internal sealed class AggregatePercentilePayload : MeterPayload
```

Comment:
The `AggregatePercentilePayload` was a new construct added for this PR because dotnet-monitor and dotnet-counters historically handled histogram events differently. Dotnet-Monitor would create a single payload that contained all of the quantiles to keep them together; I believe dotnet-counters was creating a payload for each quantile (three payloads per TraceEvent). `AggregatePercentilePayload` allows dotnet-monitor to construct a `PercentilePayload` for individual quantiles like dotnet-counters does, while still keeping the quantiles together as a unit. Let me know if that didn't make sense (or if you'd prefer a different approach).

Reply:
I think this is fine, I'd just recommend adding the explanation as comments in the code :)